providers/implementations/ciphers/ciphercommon_gcm.c

```diff
@@ -238,73 +250,76 @@ int ossl_gcm_set_ctx_params(void *vctx, const OSSL_PARAM params[])
     const OSSL_PARAM *p;
     size_t sz;
     void *vp;
+    int type, i, n;
 
     if (params == NULL)
         return 1;
 
-    ossl_scan_params_for_idx(params, &idx);
-
-    p = ossl_param_idx_const(params, &idx, PIDX_CIPHER_PARAM_AEAD_TAG);
-    if (p != NULL) {
-        vp = ctx->buf;
-        if (!OSSL_PARAM_get_octet_string(p, &vp, EVP_GCM_TLS_TAG_LEN, &sz)) {
-            ERR_raise(ERR_LIB_PROV, PROV_R_FAILED_TO_GET_PARAMETER);
-            return 0;
-        }
-        if (sz == 0 || ctx->enc) {
-            ERR_raise(ERR_LIB_PROV, PROV_R_INVALID_TAG);
-            return 0;
-        }
-        ctx->taglen = sz;
-    }
-
-    p = ossl_param_idx_const(params, &idx, PIDX_CIPHER_PARAM_AEAD_IVLEN);
-    if (p != NULL) {
-        if (!OSSL_PARAM_get_size_t(p, &sz)) {
-            ERR_raise(ERR_LIB_PROV, PROV_R_FAILED_TO_GET_PARAMETER);
-            return 0;
-        }
-        if (sz == 0 || sz > sizeof(ctx->iv)) {
-            ERR_raise(ERR_LIB_PROV, PROV_R_INVALID_IV_LENGTH);
-            return 0;
-        }
-        ctx->ivlen = sz;
-    }
-
-    p = ossl_param_idx_const(params, &idx, PIDX_CIPHER_PARAM_AEAD_TLS1_AAD);
-    if (p != NULL) {
-        if (p->data_type != OSSL_PARAM_OCTET_STRING) {
-            ERR_raise(ERR_LIB_PROV, PROV_R_FAILED_TO_GET_PARAMETER);
-            return 0;
+    n = ossl_scan_params_for_idx(params, idx);
+    for (i = 0; i < n; i++) {
+        type = ossl_param_idx_const(params, idx, i, &p);
+        if (type == -1)
```

Comment:
Is there a reason this check is necessary? Shouldn't ossl_scan_params_for_idx guarantee that given its return value n, all parsed items with indices 0 <= x < n have valid types?

Reply:
Yeah, not strictly required.  Paranoia is always good :grinning: 

I've moved this alternative over to #20935 for easier comparison.